spec/views/post_show_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#show', type: :feature do
+  before do
+    @user1 = User.create(name: 'Benjamin', photo: 'photo_url1', bio: 'I am a software developer.', postsCounter: 1)
+    @user2 = User.create(name: 'John', photo: 'photo_url2', posts_count: 0)
+    @post1 = Post.create(author: @user1, title: 'Hello', text: 'Good.', commentssCounter: 0, likesCounter: 0)
+    @comment1 = Comment.create(user: @user2, post: @post1, text: 'Great work Benjamin')
+    @like1 = Like.create(user: @user2, post: @post1)
+  end
+
+  scenario 'display the post title' do
+    visit user_post_path(@user1, @post1)
+    expect(page).to have_content(@post1.title)
+  end
+
+  scenario 'display the person who wrote the post' do
+    visit user_post_path(@user1, @post1)
+    expect(page).to have_content(@post1.author.name)
+  end
+
+  scenario 'display the post body' do
+    visit user_post_path(@user1, @post1)
+    expect(page).to have_content(@post1.text)
+  end
+
+  scenario 'display how many comments a post has' do
+    visit user_post_path(@user1, @post1)
+    expect(page).to have_content("Comments: #{@post1.commentssCounter}")
+  end
+
+  scenario 'display how many likes a post has' do
+    visit user_post_path(@user1, @post1)
```

Comment:
- [ ] Howdy! 🌟 I spotted a little challenge – all the test cases here seem to be playing hard to get and failing. 🚫🧐 Glance at the screenshot below – it's got the lowdown on the errors causing the mischief. 🔍 How about putting on your debugging cape and ironing out those wrinkles? 🦸‍♂️ Your code-taming skills will surely save the day! 🚀

![image](https://github.com/BilalLiaquat7/blogapp/assets/112261237/e2bea75d-e9f6-4aa7-b7df-4b356047b543)